shared/modules/persistence_module.go

```diff
@@ -37,6 +37,18 @@ type PersistenceModule interface {
 	HandleDebugMessage(*messaging.DebugMessage) error
 }
 
+// Tx is fed to an AtomicStore by a StoreManager to atomically apply a block to the persistence layer.
+type Tx struct {
```

Comment:
Can we not just use the block protobuf directly? I feel adding this `Tx` struct confuses with `Transaction` struct and `pgx.Tx`

Reply:
This is gone now, refactored it out cause the new approach didn't need it.